### ISSUE TITLE
error: no matching function for call to TMC2209Stepper

### ISSUE THREAD

**Issue by TheZoker** [teemuatlut/TMCStepper#56]
I'm not sure, if this is a Marlin or TMCStepper issue.

When I try to compile Marlin for my SKR E3 DIP with TMC2209, I get the following error (next to a lot of warnings):
![IMG_2019 08 01_23h30m46s_004_](https://user-images.githubusercontent.com/1368405/62328943-c7c1f180-b4b4-11e9-8eb7-f9e8854231db.png)

Here is the complete log:
https://gist.github.com/TheZoker/036b74969940e4888068b0baee3f761c

**Comment by TheZoker** [comment]
@teemuatlut Any idea about this?

**Comment by teemuatlut** [comment]
v0.4.6 should fix this when it hits PIO library manager in a few days.
If you don't want to wait, you can update your `platform.ini` to get the latest version with a direct git link.

The underlying reason is that STM32 wasn't yet defined as a Software Serial supported platform.

**Comment by TheZoker** [comment]
@teemuatlut Thanks! Now I directly tagged the master branch and plattform.io installed version 0.4.6
But now I get this error:
```
In file included from Marlin/src/module/stepper_indirection.h:47:0,
                 from Marlin/src/module/stepper.h:221,
                 from Marlin/src/Marlin.cpp:37:
.piolibdeps/BIGTREE_SKR_MINI/TMCStepper/src/TMCStepper.h:18:11: fatal error: SoftwareSerial.h: No such file or directory

************************************************************************
* Looking for SoftwareSerial.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:SoftwareSerial.h"
* Web  > https://platformio.org/lib/search?query=header:SoftwareSerial.h
*
************************************************************************

  #include <SoftwareSerial.h>
           ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

Here is my config:
https://github.com/TheZoker/Marlin/commit/dc3f0171feec3c67657cc0d491ffbd72c12b13bb

**Comment by TheZoker** [comment]
@teemuatlut Any idea where this issue comes from?

**Comment by teemuatlut** [comment]
It's trying to include a SoftwareSerial library that doesn't exist on your computer.
Maybe @GerogeFu has a suggestion which one to use.

**Comment by TheZoker** [comment]
Alright, thanks
But isn't the purpose of with dependencies (/libraries), that everything needed to build the .bin file is handled by platformio?

**Comment by teemuatlut** [comment]
Yup, but that part hasn't been specified yet. All previous platforms, pre STM, have had the SoftwareSerial library included in the framework.

**Comment by TheZoker** [comment]
Seems like this could fix the issue, right? 
https://github.com/MarlinFirmware/Marlin/commit/b76bf93a2848d309192ba477fb421c5a52a57bcf

**Comment by salacpavel** [comment]
So its just to confirm I have compiled succesfully today on very same platform, but there is "TMC CONNECTION ERROR" on screen now and M122 reports all 4 UART comm beeing down.
